### PR TITLE
Add analytics beacons to hugo and readthedocs

### PR DESCRIPTION
**Pull request by piiq**
This PR integrates the readthedocs documentation with the privacy focused Cloudflare Analytics.

The beacon is placed in the template footer and the token is being put in place as a part of the build process at readthedocs.io.

### CODE REVIEW

**Comment by piiq** [comment]
@aia I found and fixed a couple of issues with the dependencies that break readthedocs build process:

- Harmonized keras and tensorflow versions. This involved downgrading keras to 2.6 because 2.7 doesn't work with latest tf and the documentation for the prediction menu can't be built
- Added `math` and `binascii` to allowed c extensions list in `.pylintrc`
- I've noticed that the only way for the readthedocs to build documentation from `requirements-full.txt` is is the file is generated with a `--dev` flag that you were not using in the command that you've showed. So the command that generates a requirements file that works with documentation generator is:

```bash
poetry export -f requirements.txt --dev --extras prediction --without-hashes -o requirements-full.txt
```

File: requirements-full.txt [hunk 1]
```diff
@@ -190,7 +190,7 @@ pytrends==4.7.3
 pytz-deprecation-shim==0.1.0.post0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6"
 pytz==2021.3; python_full_version >= "3.6.8" and python_version >= "3.6" and python_full_version < "4.0.0"
 pyupgrade==2.18.2; python_full_version >= "3.6.1"
-pywin32==301; sys_platform == "win32" and platform_python_implementation != "PyPy" and python_version >= "3.6"
+pywin32==302; sys_platform == "win32" and platform_python_implementation != "PyPy" and python_version >= "3.6"
```
Comment (aia) [inline requirements-full.txt:193]: I keep running into issues with pywin32. I think we should keep it at 301.

Comment (aia) [reply requirements-full.txt:193]: actually 302 is working again for me now. Windows inconsistencies are annoying.

Comment (DidierRLopes) [reply requirements-full.txt:193]: Is this good to go then?

Comment (piiq) [reply requirements-full.txt:193]: It is to me=) @aia ?

**Review by DidierRLopes — APPROVED** [review]

**Comment by piiq** [comment]
I’ve updated the PR to include adding a beacon to the Hugo web-site. The token is already added to the repo secrets so we should see something as soon as we merge this.

**Review by DidierRLopes — APPROVED** [review]

**Review by aia — APPROVED** [review]

**Pull request merged**